src/main/java/com/cloudant/client/org/lightcouch/CouchDbClientBase.java

```diff
@@ -330,64 +273,92 @@ public HttpResponse executeRequest(HttpRequestBase request) {
      * @return {@link Response}
      */
     Response delete(URI uri) {
-        HttpResponse response = null;
-        try {
-            HttpDelete delete = new HttpDelete(uri);
-            response = executeRequest(delete);
-            return getResponse(response, Response.class, getGson());
-        } finally {
-            close(response);
-        }
+        HttpConnection connection = Http.DELETE(uri);
+        return executeRequest(connection, Response.class);
     }
 
-
     /**
      * Performs a HTTP GET request.
      *
-     * @return {@link InputStream}
+     * @return Input stream with response
      */
-    InputStream get(HttpGet httpGet) {
-        HttpResponse response = executeRequest(httpGet);
-        return getStream(response);
+    public InputStream get(URI uri) {
+        HttpConnection httpConnection = Http.GET(uri);
+        return executeRequest(httpConnection);
     }
 
     /**
      * Performs a HTTP GET request.
      *
-     * @return {@link InputStream}
+     * @return Class type of object T (i.e. {@link Response}
      */
-    InputStream get(URI uri) {
-        HttpGet get = new HttpGet(uri);
-        get.addHeader("Accept", "application/json");
-        return get(get);
+    public <T> T get(URI uri, Class<T> classType) {
+        HttpConnection connection = Http.GET(uri);
+        if(classType.isInstance(HttpConnection.class)) {
+            return (T)connection;
+        } else {
+            //return executeToJsonObject(connection, classType);
+            return executeRequest(connection, classType);
+        }
+
     }
 
     /**
-     * Performs a HTTP GET request.
+     * Performs a HTTP HEAD request.
      *
-     * @return An object of type T
+     * @return {@link Response}
      */
-    public <T> T get(URI uri, Class<T> classType) {
-        InputStream in = null;
-        try {
-            in = get(uri);
-            return getGson().fromJson(new InputStreamReader(in, "UTF-8"), classType);
-        } catch (UnsupportedEncodingException e) {
-            // This should never happen as every implementation of the java platform is required
-            // to support UTF-8.
-            throw new RuntimeException(e);
-        } finally {
-            close(in);
+    InputStream head(URI uri) {
+        HttpConnection connection = Http.HEAD(uri);
+        return executeRequest(connection);
+    }
+
+    /**
+     * Performs a HTTP PUT request, saves or updates a document.
+     *
+     * @return Input stream with response
+     */
+    InputStream put(URI uri) {
+        return put(uri, null);

```

Comment:
Improve javadoc to say this defaults to "application/json" content type and then pass "application/json" String, instead of `null`.